packages/grid/x-grid/src/tests/selection.XGrid.test.tsx

```diff
@@ -138,6 +138,22 @@ describe('<XGrid /> - Selection', () => {
     expect(getSelectedRows(apiRef)).to.deep.equal([0]);
   });
 
+  it('should call onSelectionModelChange when selection state changes', () => {
+    const onSelectionModelChange = spy();
+    const { setProps } = render(
+      <Test selectionModel={[0]} onSelectionModelChange={onSelectionModelChange} />,
+    );
+    setProps({
+      rows: [
+        {
+          id: 1,
+          brand: 'Nike',
+        },
+      ],
+    });
+    expect(onSelectionModelChange.callCount).to.equal(2);
```

Comment:
Could we also assert that the callback value is correct?

Reply:
To leave as a comment or to assert right after the first render?

```suggestion
    // TODO should only fire once
    expect(onSelectionModelChange.callCount).to.equal(2);
```

Reply:
Just to be sure - the correct behavior in the case of server side pagination will be an empty array for the `selectionModel` because the selection state will be empty (there won't be any matching ids).
I'm asking because there is already a test that is asserting that new rows have the same ids they will remain selected - https://github.com/mui-org/material-ui-x/blob/master/packages/grid/x-grid/src/tests/selection.XGrid.test.tsx#L127

Reply:
Updated